### PR TITLE
Ensure all required health checks are run

### LINKED ISSUE #52: Greenboot stops checking required health checks if one fails
As $subject states, this behaviour makes sense as Greenboot would mark the boot as "red/faulty" anyway, but it would bring more value to the user knowing all the required health checks that had failed, not just the first one.

### PR DESCRIPTION
**Pull request by jnogol**
Closes #52

### CODE REVIEW

File: usr/libexec/greenboot/greenboot [hunk 1]
```diff
@@ -7,6 +7,7 @@ script_runner () {
   local scripts_dir=$1; shift
   local mode=$1; shift
   local start_msg=$1; shift
+  local required_hc_failed="false"
```
Comment (LorbusChris) [inline usr/libexec/greenboot/greenboot:10]: I don't think quotation marks are necessary here

Comment (jnogol) [reply usr/libexec/greenboot/greenboot:10]: You're right. Pushing changes...

File: usr/libexec/greenboot/greenboot [hunk 2]
```diff
@@ -19,13 +20,17 @@ script_runner () {
...
+          required_hc_failed=true
+          echo "<0>$failure_msg. Continuing..." >&2
       esac
     else
       echo "<6>Script '$(basename "$script")' SUCCESS"
     fi
   done
+
+  if [[ $required_hc_failed == true ]]; then
+    exit 1
```
Comment (LorbusChris) [inline usr/libexec/greenboot/greenboot:32]: maybe echo something here, too, noting some (at least one) tests have failed?

Comment (jnogol) [reply usr/libexec/greenboot/greenboot:32]: Good idea.

Comment (jnogol) [reply usr/libexec/greenboot/greenboot:32]: What do you think @LorbusChris?

journalctl:
```
Oct 05 08:26:14 fedora 00_required_scripts_start.sh[681]: Running greenboot Required Health Check Scripts
Oct 05 08:26:14 fedora greenboot[670]: Script '00_required_scripts_start.sh' SUCCESS
Oct 05 08:26:14 fedora greenboot[670]: Script '01_fail.sh' FAILURE (exit code '1'). Continuing...
Oct 05 08:26:15 fedora greenboot[670]: Script '02_fail.sh' FAILURE (exit code '1'). Continuing...
Oct 05 08:26:15 fedora zsh.sh[694]: ZSH is available
Oct 05 08:26:15 fedora greenboot[670]: Script 'zsh.sh' SUCCESS
Oct 05 08:26:15 fedora greenboot[670]: ATTENTION: At least one required health check has failed
```

MOTD:
```
Script '01_fail.sh' FAILURE (exit code '1'). Continuing...
Script '02_fail.sh' FAILURE (exit code '1'). Continuing...
ATTENTION: At least one required health check has failed
Boot Status is RED - Health Check FAILURE!
```

If OK, will commit and update the PR.

Comment (LorbusChris) [reply usr/libexec/greenboot/greenboot:32]: LGTM, although attention would sound like a warning to me, not an error. Not too important here though I think.

Comment (jnogol) [reply usr/libexec/greenboot/greenboot:32]: Yeah, I wasn't too sure about that choice of words either but couldn't think of a better one. @nullr0ute, any feedback?

Comment (nullr0ute) [reply usr/libexec/greenboot/greenboot:32]: We should possibly also report which of the health checks failed so the user knows where to look.

Comment (LorbusChris) [reply usr/libexec/greenboot/greenboot:34]: That should already be the case here (e.g. `Script '02_fail.sh' FAILURE (exit code '1'). Continuing...` in the MOTD above)

**Pull request merged**